commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/StartUpActionsParser.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ * Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.util;
+
+import com.google.gwt.user.client.Window;
+
+import org.eclipse.che.ide.util.loging.Log;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility class that parse IDE URL and get start up parameters form it according specific rules.
+ * URL for parsing comes from @see Window.Location.
+ *
+ * For example:
+ * after parsing URL like it http://localhost:8080/ide/dev6?action=createProject:projectName=test;projectType=maven
+ * we will get list with one action (@see StartUpAction) with ID 'createProject' and two parameters
```

Comment:
The same as there https://github.com/codenvy/che-core/pull/618#discussion_r49584748